sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -145,9 +147,11 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry)
    * @param returnDataType return type of udf. If it is null, spark would try to infer
    *                       via reflection.
    */
-  private[sql] def registerJava(name: String, className: String, returnDataType: DataType): Unit = {
+  def registerJava(name: String, className: String, returnDataType: DataType): Unit = {
```

Comment:
I have to make this method public so I can call it from REPL.

Reply:
I am not against this. I am trying to understand the user facing consequences though. I'd probably prefer that we add support for Scala UDFs as well. That can be done in a follow-up though.

Reply:
Can you file a follow-up?

Reply:
Will do.